Network Observability 1.4 Release Notes (#63934)
openshift/openshift-docs: networking/network_observability/network-observability-operator-release-notes.adoc

For more information, see xref:../../networking/network_observability/observing-network-traffic.adoc#network-observability-quickfilter_nw-observe-network-traffic[Configuring Quick Filters].
 
This release introduces the following configuration enhancements:

* The `spec.processor.clusterName` parameter is added so that the name of the cluster appears in the flows data. This is useful in a multi-cluster context. When using {product-title}, leave empty to make it automatically determined.
* You now have the option to specify different namespaces for all Network Observability components using the `spec.loki.tls.caCert.namespace` and `spec.exporters.kafka.tls.caCert.namespace` fields.

For more information, see xref:../../networking/network_observability/configuring-operator.adoc#network-observability-flowcollector-view_network_observability[Flow Collector sample resource] and xref:../../networking/network_observability/flowcollector-api.adoc#network-observability-flowcollector-api-specifications_network_observability[Flow Collector API Reference].

[id="network-observability-dns-tracking-1.4"]
==== DNS tracking

> In 1.4, you can enable DNS tracking with eBPF tracepoint hooks. With Network Observability DNS tracking, you can monitor your network, conduct security analysis, and troubleshoot DNS issues in the *Network Traffic* and *Overview* views in the web console. 

Author: I think I may have this _slightly_ backwards...
>  you can enable DNS tracking with eBPF tracepoint hooks

should be

> The Network Observability Operator makes use of eBPF tracepoint hooks to enable DNS tracking

Yes? No?

Reviewer: yes, +1, so the userd aren't misguided into thinking they would have to do fancy eBPF stuff themselves :-)

Author: Addressed! TY!

* Previously, during spikes of network traffic load, certain eBPF pods were OOM-killed and go into `CrashLoopBackOff` state with the 429 error: 'Per stream rate limit exceeded (limit:xMB/sec) while attempting to ingest for stream'. Now, you can xref:../../networking/network_observability/troubleshooting-network-observability.adoc#network-observability-troubleshooting-loki-tenant-rate-limit_network-observability-troubleshooting[configure the `LokiStack` custom resource `spec.limits.global.ingestion`] with the `perStreamRateLimit` and `perStreamRateLimitBurst` specifications restart the pods in your cluster and stop the 429 rate-limit error. (link:https://issues.redhat.com/browse/NETOBSERV-975[*NETOBSERV-975*]) 

[id="network-observability-operator-1.4.0-known-issues"]
=== Known issue

> * Since the 1.2.0 release of the Network Observability Operator, using Loki Operator 5.6, a Loki certificate change periodically affects the `flowlogs-pipeline` pods and results in dropped flows rather than flows written to Loki. The problem self-corrects after some time, but it still causes temporary flow data loss during the Loki certificate change. (link:https://issues.redhat.com/browse/NETOBSERV-980[*NETOBSERV-980*])

Reviewer 2: Same comment as below

* Previously, during spikes of network traffic load, certain eBPF pods were OOM-killed and go into `CrashLoopBackOff` state with the 429 error: 'Per stream rate limit exceeded (limit:xMB/sec) while attempting to ingest for stream'. Now, you can xref:../../networking/network_observability/troubleshooting-network-observability.adoc#network-observability-troubleshooting-loki-tenant-rate-limit_network-observability-troubleshooting[configure the `LokiStack` custom resource `spec.limits.global.ingestion`] with the `perStreamRateLimit` and `perStreamRateLimitBurst` specifications restart the pods in your cluster and stop the 429 rate-limit error. (link:https://issues.redhat.com/browse/NETOBSERV-975[*NETOBSERV-975*]) 

[id="network-observability-operator-1.4.0-known-issues"]
=== Known issue

> * Since the 1.2.0 release of the Network Observability Operator, using Loki Operator 5.6, a Loki certificate change periodically affects the `flowlogs-pipeline` pods and results in dropped flows rather than flows written to Loki. The problem self-corrects after some time, but it still causes temporary flow data loss during the Loki certificate change. This issue has only been observed in large-scale environments of 120 nodes or greater. (link:https://issues.redhat.com/browse/NETOBSERV-980[*NETOBSERV-980*])

Reviewer 3: should we list [DNS TCP issue](https://issues.redhat.com/browse/NETOBSERV-1245) under known issue? it was reported as Bug but later converted as story given the scope. I feel it'd be useful for customer to learn about this limitation with DNSTracking feature cc @msherif1234

Reviewer 3: also, how about listing mDNS also something not supported yet?

Author: @memodi I don't see mDNS listed in the API as "unsupported"...I guess usually we reserve those unsupported mentions for "unsupported but still works at your own risk" type things. For example, IPFIX was listed as "unsupported" for a while. My feeling is we don't usually mention unsupported things in the release notes, unless its a deprecation notice.

Reviewer 4: yeah we need to say no mDNS, no DNS over TCP and no DNS-SD "service discovery"

Reviewer 3: @skrthomas - this isn't about something that already existed, but more about other DNS features that may imply to be working/supported while we know with 1.4 they do not work (existing reported issues), so would be better to list them.

Author: Ok i added two known issues for https://issues.redhat.com/browse/NETOBSERV-1304 and https://issues.redhat.com/browse/NETOBSERV-1245. Thanks for adding the RN Text n the jiras @msherif1234

[id="network-observability-without-loki-1.4"]
==== Network Observability without Loki
You can now install the Network Observability Operator without Loki and only export flows to KAFKA or IPFIX. For more information, see xref:../../networking/network_observability/installing-operators.adoc#network-observability-without-loki_network_observability[Network Observability without Loki]. 

[id="network-observability-enhanced-configuration-and-ui-1.4"]
==== Notable enhancements
The 1.4 release of the Network Observability Operator adds improvements and new capabilities to the {product-title} web console plugin and the Operator configuration. 

This release introduces the following web console enhancements:

* In the *Query Options*, the *Duplicate flows* checkbox is added to choose whether or not to show duplicated flows.
* You can now filter source and destination traffic with image:arrow-up-long-solid.png[,10] *One-way*, image:arrow-up-long-solid.png[,10] image:arrow-down-long-solid.png[,10] *Back-and-forth*, and *Swap* filters. 
* The Network Observability metrics dashboards in *Observe* -> *Dashboard* -> *NetObserv* and *NetObserv / Health* are modified as follows:

> ** The *NetObserv* dashboard shows top bytes, packets sent, packets received per nodes, namespaces, workload, and pods. Flow graphs are removed from this dashboard.

Reviewer 5: ```suggestion
** The *NetObserv* dashboard shows top bytes, packets sent, packets received per nodes, namespaces and workload. Flow graphs are removed from this dashboard. *Infrastructure* and *Application* metrics are shown in a split-view for namespace and workloads.
```

> ** The *NetObserv / Health* dashboard shows flows overhead as well as top flow rates per nodes, namespaces, workload, and pods.

Reviewer 5: ```suggestion
** The *NetObserv / Health* dashboard shows flows overhead as well as top flow rates per nodes, namespaces and workload.
```

> ** Infrastructure and Application metrics are shown in a split-view for namespace, pod, and workloads. 

Reviewer 5: ```suggestion
** Infrastructure and Application metrics are shown in a split-view for namespace and workloads. 
```

Author: @jpinsonneau On all the namespace, workload, node, is it ok for all these to be plural, like namespaces, workloads, and nodes.

Reviewer 5: yes sure :+1:

==== DNS tracking
In 1.4, you can enable DNS tracking with eBPF tracepoint hooks. With Network Observability DNS tracking, you can monitor your network, conduct security analysis, and troubleshoot DNS issues in the *Network Traffic* and *Overview* views in the web console. 

For more information, see xref:../../networking/network_observability/observing-network-traffic#network-observability-dns-overview_nw-observe-network-traffic[Configuring DNS tracking] and xref:../../networking/network_observability/observing-network-traffic#network-observability-dns-tracking_nw-observe-network-traffic[Working with DNS tracking].

//Packet drops needs separate RN PR that doesn't cherrypick to 4.10+ since its only supported in 4.13+. This PR will go to 4.10+
[id="SR-IOV-configuration-1.4"]
==== SR-IOV support
You can now collect traffic from a cluster with Single Root I/O Virtualization (SR-IOV) device. For more information, see xref:../../networking/network_observability/configuring-operator.adoc#network-observability-SR-IOV-config_network_observability[Configuring the monitoring of SR-IOV interface traffic].

[id="IPFIX-support-1.4"]
==== IPFIX exporter support

> You can now export eBPF-enriched network flows to IPFIX. For more information, see xref:../../networking/network_observability/configuring-operator.adoc#network-observability-enriched-flows_network_observability[Export enriched network flow data].

Reviewer 5: ```suggestion
You can now export eBPF-enriched network flows to IPFIX collector. For more information, see xref:../../networking/network_observability/configuring-operator.adoc#network-observability-enriched-flows_network_observability[Export enriched network flow data].
```

For an overview of the Network Observability Operator, see xref:../../networking/network_observability/network-observability-overview.adoc#dependency-network-observability[About Network Observability Operator].

[id="network-observability-operator-release-notes-1-4"]
== Network Observability Operator 1.4.0
The following advisory is available for the Network Observability Operator 1.4.0:

* https://errata.devel.redhat.com/advisory/116729
 https://access.redhat.com/errata/RHSA-2023:1817[RHSA-2023:1817 Network Observability Operator 1.4.0]

[id="network-observability-channel-deprecation-1.4"]

> === Channel Deprecation

Reviewer: Do you make a difference in doc between "deprecation" and "removal"? I'm used to seeing "deprecation" as something like in a grace period before removal; which, if I'm seeing it correctly, wouldn't be the case now, it's really removed.

Author: Ah yeah this is corect. I see have guidance about the stages: https://redhat-documentation.github.io/supplementary-style-guide/#_deprecated_functionality

This is my first time actually going through a removal of any kind of feature, over time, so this is new to me.

> You must switch your channel from `v1.0.x` to `stable` to receive the latest Operator updates. The `v1.0.x` channel is now deprecated.

Reviewer: same as above?

```suggestion
You must switch your channel from `v1.0.x` to `stable` to receive the latest Operator updates. The `v1.0.x` channel is now removed.
```

* The Network Observability metrics dashboards in *Observe* -> *Dashboard* -> *NetObserv* and *NetObserv / Health* are modified as follows:
** The *NetObserv* dashboard shows top bytes, packets sent, packets received per nodes, namespaces, workload, and pods. Flow graphs are removed from this dashboard.
** The *NetObserv / Health* dashboard shows flows overhead as well as top flow rates per nodes, namespaces, workload, and pods.
** Infrastructure and Application metrics are shown in a split-view for namespace, pod, and workloads. 
  

For more information, see xref:../../networking/network_observability/observing-network-traffic.adoc#network-observability-quickfilter_nw-observe-network-traffic[Configuring Quick Filters] and xref:../../networking/network_observability/network-observability-overview.adoc#network-observability-dashboards[Network Observability metrics].
 
This release introduces the following configuration enhancements:

* The `spec.processor.clusterName` parameter is added so that the name of the cluster appears in the flows data. This is useful in a multi-cluster context. When using {product-title}, leave empty to make it automatically determined.

> * You now have the option to specify different namespaces for all Network Observability components using the `spec.loki.tls.caCert.namespace` and `spec.exporters.kafka.tls.caCert.namespace` fields.

Reviewer: there's actually more places than loki.tls.caCert and exporters.kafka.tls.caCert, it seems an overkill to list them all, what about saying something like "You now have the option to specify different namespaces for any configured ConfigMap or Secret reference, such as in certificates configuration" ?

Author: Okay staying more general here makes sense. I was also thinking that the UI enhancements section may be overkill.

[id="network-observability-dns-tracking-1.4"]
==== DNS tracking

> In 1.4, the Network Observability Operator makes use of eBPF tracepoint hooks to enable DNS tracking DNS tracking. You can monitor your network, conduct security analysis, and troubleshoot DNS issues in the *Network Traffic* and *Overview* views in the web console. 

Reviewer 2: ```suggestion
In 1.4, the Network Observability Operator makes use of eBPF tracepoint hooks to enable DNS tracking. You can monitor your network, conduct security analysis, and troubleshoot DNS issues in the *Network Traffic* and *Overview* views in the web console. 
```

Author: Copy/paste error on my part. I _just_ reworded this and left the duplicate DNS tracking 🤦‍♀️

* Previously, the Network Observability Operator agents were not able to capture traffic on network interfaces when configured with Multus or SR-IOV, non-default network namespaces. Now, all available network namespaces are recognized and used for capturing flows, allowing capturing traffic for SR-IOV. There are xref:../../networking/network_observability/configuring-operator.adoc#network-observability-SR-IOV-config_network_observability[configurations needed] for the `FlowCollector` and `SRIOVnetwork` custom resource to collect traffic. 
(link:https://issues.redhat.com/browse/NETOBSERV-1283[*NETOBSERV-1283*])

* Previously, in the Network Observability Operator details from *Operators* -> *Installed Operators*, the `FlowCollector` *Status* field might have reported incorrect information about the state of the deployment. The status field now shows the proper conditions with improved messages. The history of events is kept, ordered by event date. (link:https://issues.redhat.com/browse/NETOBSERV-1224[*NETOBSERV-1224*])

> * Previously, during spikes of network traffic load, certain eBPF pods were OOM-killed and go into `CrashLoopBackOff` state with the 429 error: 'Per stream rate limit exceeded (limit:xMB/sec) while attempting to ingest for stream'. Now, you can xref:../../networking/network_observability/troubleshooting-network-observability.adoc#network-observability-troubleshooting-loki-tenant-rate-limit_network-observability-troubleshooting[configure the `LokiStack` custom resource `spec.limits.global.ingestion`] with the `perStreamRateLimit` and `perStreamRateLimitBurst` specifications restart the pods in your cluster and stop the 429 rate-limit error. (link:https://issues.redhat.com/browse/NETOBSERV-975[*NETOBSERV-975*]) 

Reviewer: I think this is mixing several things: the "429 error: Per stream rate limit exceeded" isn't thrown by the ebpf agents - and the new LokiStack config was tracked in https://issues.redhat.com/browse/NETOBSERV-717 which we didn't include in the release note because it's a loki-operator change, not netobserv.
We may reformulate to focus strictly on the eBPF agent memory improvement that we did to avoid OOM-killing / CrashLoopBackOff

Author: Ah ok...I'm not sure how I got this mixed up but thanks for clarifying the difference.

Reviewer: probably because the original JIRA ticket was also mixing those things :)

* Previously when `processor.metrics.tls` was set to `PROVIDED` the `insecureSkipVerify` option value was forced to be `true`. Now you can set `insecureSkipVerify` to `true` or `false`, and provide a CA certificate if needed.  (link:https://issues.redhat.com/browse/NETOBSERV-1087[NETOBSERV-1087])

[id="network-observability-operator-1.4.0-known-issues"]

> === Known issue

Reviewer: ```suggestion
=== Known issues
```

* Since the 1.2.0 release of the Network Observability Operator, using Loki Operator 5.6, a Loki certificate change periodically affects the `flowlogs-pipeline` pods and results in dropped flows rather than flows written to Loki. The problem self-corrects after some time, but it still causes temporary flow data loss during the Loki certificate change. This issue has only been observed in large-scale environments of 120 nodes or greater. (link:https://issues.redhat.com/browse/NETOBSERV-980[*NETOBSERV-980*])

* Currently, larger DNS packets require the eBPF agent to look for DNS header outside of the 1st skb segment. new ebpf agent helper function need to implemented to support it. Currently, there is no workaround for this issue. (link:https://issues.redhat.com/browse/NETOBSERV-1304[*NETOBSERV-1304*])

> * Currently, DNS over TCP packets require the eBPF agent to look for DNS header outside of the 1st skb segment. new ebpf agent helper function need to implemented to support it. Currently, there is no workaround for this issue. (link:https://issues.redhat.com/browse/NETOBSERV-1245[*NETOBSERV-1245*])

Reviewer: For these two, you might clarify that this is only affecting the DNS tracking feature, ie. when `spec.agent.ebpf.features` includes `DNSTracking` - so that users not using it aren't worried

* Currently, when `spec.agent.ebpf.features` includes DNSTracking, larger DNS packets require the eBPF agent to look for DNS header outside of the 1st socket buffer (SKB) segment. A new eBPF agent helper function needs to be implemented to support it. Currently, there is no workaround for this issue. (link:https://issues.redhat.com/browse/NETOBSERV-1304[*NETOBSERV-1304*])

* Currently, when `spec.agent.ebpf.features` includes DNSTracking, DNS over TCP packets requires the eBPF agent to look for DNS header outside of the 1st SKB segment. A new eBPF agent helper function needs to be implemented to support it. Currently, there is no workaround for this issue. (link:https://issues.redhat.com/browse/NETOBSERV-1245[*NETOBSERV-1245*])

> * Currently, when using connection tracking with a Kafka deployment that might impact metrics related to connection tracking. There is no workaround for this issue. However, to avoid issues, it is recommended not to use conversation tracking when `deploymentModel` is set to `KAFKA`. (link:https://issues.redhat.com/browse/NETOBSERV-926[*NETOBSERV-926*])

Author: Ah, yeah the connection vs. conversation wording. Ok, I amended this note to what you suggest. TY!

For an overview of the Network Observability Operator, see xref:../../networking/network_observability/network-observability-overview.adoc#dependency-network-observability[About Network Observability Operator].

[id="network-observability-operator-release-notes-1-4"]
== Network Observability Operator 1.4.0
The following advisory is available for the Network Observability Operator 1.4.0:

> * link:https://errata.devel.redhat.com/advisory/2023:116729[RHSA-2023:1817 Network Observability Operator 1.4.0]

Author: I do not know the live ID yet...I don't see it in the errata advisory.

> * link:https://errata.devel.redhat.com/advisory/2023:116729[RHSA-2023:1817 Network Observability Operator 1.4.0]

Author: Got the live ID!

[id="network-observability-channel-removal-1.4"]
=== Channel removal
You must switch your channel from `v1.0.x` to `stable` to receive the latest Operator updates. The `v1.0.x` channel is now removed.

[id="network-observability-operator-1.4.0-features-enhancements"]
=== New features and enhancements

[id="network-observability-without-loki-1.4"]
==== Network Observability without Loki
The Network Observability Operator is now functional and usable without Loki. If Loki is not installed, it can only export flows to KAFKA or IPFIX format and provide metrics in the Network Observability metrics dashboards. For more information, see xref:../../networking/network_observability/installing-operators.adoc#network-observability-without-loki_network_observability[Network Observability without Loki]. 

[id="network-observability-enhanced-configuration-and-ui-1.4"]

> ==== Notable enhancements

Reviewer 6: This may not be something you want to change, but I was thinking maybe you want to put the notable enhancements before the general new features and enhancements. You could adjust the headings accordingly if you wanted to do that too. So,

> [id="network-observability-operator-1.4.0-features-enhancements"]
> === New features and enhancements
> [id="network-observability-enhanced-configuration-and-ui-1.4"]
> ==== Notable enhancements
> [id="network-observability-without-loki-1.4"]
> ===== Network Observability without Loki

I think what I am trying to get at is the entry "Network Observability without Loki" seems out of place where it is.

Author: Yeah I can totally see that now that you point it out. I like the reorg, suggestion taken :)

The 1.4 release of the Network Observability Operator adds improvements and new capabilities to the {product-title} web console plugin and the Operator configuration. 

> This release introduces the following web console enhancements:

Reviewer 6: I was looking at the screen recording of the preview and you might want to consider making this sentence and the one about configuration enhancements a discrete heading or something like that. I feel like they kind of get buried in the text. It is totally up to you though.

Author: Ooo discrete heading for these is a great suggestion! TY!!

** Infrastructure and Application metrics are shown in a split-view for namespaces and workloads. 
  

For more information, see xref:../../networking/network_observability/observing-network-traffic.adoc#network-observability-quickfilter_nw-observe-network-traffic[Configuring Quick Filters] and xref:../../networking/network_observability/network-observability-overview.adoc#network-observability-dashboards[Network Observability metrics].
 
This release introduces the following configuration enhancements:

* You now have the option to specify different namespaces for any configured ConfigMap or Secret reference, such as in certificates configuration.

For more information, see xref:../../networking/network_observability/configuring-operator.adoc#network-observability-flowcollector-view_network_observability[Flow Collector sample resource] and xref:../../networking/network_observability/flowcollector-api.adoc#network-observability-flowcollector-api-specifications_network_observability[Flow Collector API Reference].

[id="network-observability-dns-tracking-1.4"]
==== DNS tracking

> In 1.4, the Network Observability Operator makes use of eBPF tracepoint hooks to enable DNS tracking. You can monitor your network, conduct security analysis, and troubleshoot DNS issues in the *Network Traffic* and *Overview* views in the web console. 

Reviewer 6: ```suggestion
In 1.4, the Network Observability Operator makes use of eBPF tracepoint hooks to enable DNS tracking. You can monitor your network, conduct security analysis, and troubleshoot DNS issues in the *Network Traffic* and *Overview* pages in the web console. 
```
https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#page

* Previously when `processor.metrics.tls` was set to `PROVIDED` the `insecureSkipVerify` option value was forced to be `true`. Now you can set `insecureSkipVerify` to `true` or `false`, and provide a CA certificate if needed.  (link:https://issues.redhat.com/browse/NETOBSERV-1087[NETOBSERV-1087])

[id="network-observability-operator-1.4.0-known-issues"]
=== Known issues
* Since the 1.2.0 release of the Network Observability Operator, using Loki Operator 5.6, a Loki certificate change periodically affects the `flowlogs-pipeline` pods and results in dropped flows rather than flows written to Loki. The problem self-corrects after some time, but it still causes temporary flow data loss during the Loki certificate change. This issue has only been observed in large-scale environments of 120 nodes or greater. (link:https://issues.redhat.com/browse/NETOBSERV-980[*NETOBSERV-980*])

> * Currently, when `spec.agent.ebpf.features` includes DNSTracking, larger DNS packets require the eBPF agent to look for DNS header outside of the 1st socket buffer (SKB) segment. A new eBPF agent helper function needs to be implemented to support it. Currently, there is no workaround for this issue. (link:https://issues.redhat.com/browse/NETOBSERV-1304[*NETOBSERV-1304*])

Reviewer 6: I think eBPF should be in backticks

* Currently, when `spec.agent.ebpf.features` includes DNSTracking, DNS over TCP packets requires the eBPF agent to look for DNS header outside of the 1st SKB segment. A new eBPF agent helper function needs to be implemented to support it. Currently, there is no workaround for this issue. (link:https://issues.redhat.com/browse/NETOBSERV-1245[*NETOBSERV-1245*])

> * Currently, when using a Kafka deployment model, if conversation tracking is configured, conversation events might be duplicated across Kafka consumers, resulting in inconsistent tracking of conversations, and incorrect volumetric data. For that reason, it is not recommended to configure conversation tracking when `deploymentModel` is set to `KAFKA`. (link:https://issues.redhat.com/browse/NETOBSERV-926[*NETOBSERV-926*])

Reviewer 6: I see KAFKA in backticks here but above it is not in backticks. Should both be in backticks?

Author: Yes, when I'm referencing the deployment model spec it should.

> * Currently, when the `processor.metrics.server.tls.type` is configured to use a `PROVIDED` certificate, the operator enters an unsteady state that might affect its performances and resource consumption. It is recommended to not use a `PROVIDED` certificate until this issue is resolved, and instead using an auto-generated certificate, setting `processor.metrics.server.tls.type` to `AUTO`. (link:https://issues.redhat.com/browse/NETOBSERV-1293)[*NETOBSERV-1293*]

Reviewer 6: ```suggestion
* Currently, when the `processor.metrics.server.tls.type` is configured to use a `PROVIDED` certificate, the operator enters an unsteady state that might affect its performance and resource consumption. It is recommended to not use a `PROVIDED` certificate until this issue is resolved, and instead using an auto-generated certificate, setting `processor.metrics.server.tls.type` to `AUTO`. (link:https://issues.redhat.com/browse/NETOBSERV-1293)[*NETOBSERV-1293*]
```

* Previously, the SCTP, ICMPv4 and ICMPv6 protocols were not covered by the Network Observability agents, resulting in a less comprehensive network flows coverage. These protocols are now recognized to improve the flows coverage. (link:https://issues.redhat.com/browse/NETOBSERV-934[*NETOBSERV-934*])

[id="network-observability-operator-1.3.0-known-issues"]
=== Known issues
* When `processor.metrics.tls` is set to `PROVIDED` in the `FlowCollector`, the `flowlogs-pipeline` `servicemonitor` is not adapted to the TLS scheme. (link:https://issues.redhat.com/browse/NETOBSERV-1087[NETOBSERV-1087])

> * Since the 1.2.0 release of the Network Observability Operator, using Loki Operator 5.6, a Loki certificate transition periodically affects the `flowlogs-pipeline` pods and results in dropped flows rather than flows written to Loki. The problem self-corrects after some time, but it still causes temporary flow data loss during the Loki certificate transition. (link:https://issues.redhat.com/browse/NETOBSERV-980[*NETOBSERV-980*])

Reviewer 2: Might be worth mentioning this has only been observed in large-scale environments (e.g. 120 nodes in our QE test)

* Previously, the SCTP, ICMPv4 and ICMPv6 protocols were not covered by the Network Observability agents, resulting in a less comprehensive network flows coverage. These protocols are now recognized to improve the flows coverage. (link:https://issues.redhat.com/browse/NETOBSERV-934[*NETOBSERV-934*])

[id="network-observability-operator-1.3.0-known-issues"]
=== Known issues

> * When `processor.metrics.tls` is set to `PROVIDED` in the `FlowCollector`, the `flowlogs-pipeline` `servicemonitor` is not adapted to the TLS scheme. (link:https://issues.redhat.com/browse/NETOBSERV-1087[NETOBSERV-1087])

Author: Rather than removing it from here, I think that we would need to document this as a bug fix for 1.4 then. Because it was indeed a Known Issue at the 1.3 release, I don't think we can remove it. But we definitely should document that it is now fixed since it was a published known issue. 

@OlivierCazade can you please add a Release Note text for this bug: https://issues.redhat.com/browse/NETOBSERV-1087?

> * When `processor.metrics.tls` is set to `PROVIDED` in the `FlowCollector`, the `flowlogs-pipeline` `servicemonitor` is not adapted to the TLS scheme. (link:https://issues.redhat.com/browse/NETOBSERV-1087[NETOBSERV-1087])

Author: I just need to know the basic "What happens now?". The Bug fix will be like this:
Previously, when `processor.metrics.tls` was set to `PROVIDED` in the `FlowCollector`, the `flowlogs-pipeline` `servicemonitor` did not adapt to the TLS scheme. Now, <what I need to know>.

* Previously, the SCTP, ICMPv4 and ICMPv6 protocols were not covered by the Network Observability agents, resulting in a less comprehensive network flows coverage. These protocols are now recognized to improve the flows coverage. (link:https://issues.redhat.com/browse/NETOBSERV-934[*NETOBSERV-934*])

[id="network-observability-operator-1.3.0-known-issues"]
=== Known issues
* When `processor.metrics.tls` is set to `PROVIDED` in the `FlowCollector`, the `flowlogs-pipeline` `servicemonitor` is not adapted to the TLS scheme. (link:https://issues.redhat.com/browse/NETOBSERV-1087[NETOBSERV-1087])

> * Since the 1.2.0 release of the Network Observability Operator, using Loki Operator 5.6, a Loki certificate transition periodically affects the `flowlogs-pipeline` pods and results in dropped flows rather than flows written to Loki. The problem self-corrects after some time, but it still causes temporary flow data loss during the Loki certificate transition. This issue has only been observed in large-scale environments of 120 nodes or greater.(link:https://issues.redhat.com/browse/NETOBSERV-980[*NETOBSERV-980*])

Author: Ah good catch. I think this is also commonly referred to as certificate rotation, which is used elsewhere in the doc. I will change this.

> * Since the 1.2.0 release of the Network Observability Operator, using Loki Operator 5.6, a Loki certificate transition periodically affects the `flowlogs-pipeline` pods and results in dropped flows rather than flows written to Loki. The problem self-corrects after some time, but it still causes temporary flow data loss during the Loki certificate transition. This issue has only been observed in large-scale environments of 120 nodes or greater.(link:https://issues.redhat.com/browse/NETOBSERV-980[*NETOBSERV-980*])

Author: I use "change" in another place actually so I'm going to use change.
